services/TipiSuper.py

```diff
@@ -14,11 +16,43 @@
             while True:
                 time.sleep(20)
         
+    # reboot the PI
     if os.path.exists("/tmp/tipireboot"):
         callargs = ["/sbin/shutdown", "-r", "now"]
         if call(callargs) != 0:
             raise Exception("failed to run /sbin/shutdown")
         else:
             while True:
                 time.sleep(20)
+
+    # Configure WiFi
+    wificonfig="/tmp/wificonfig"
+    if os.path.exists(wificonfig):
+        # wait a couple seconds so file will be complete.
+        time.sleep(2)
+        # expect two config entries
+        try:
+            # read the username and password
+            with open(wificonfig) as fh:
+                ssid = fh.readline().rstrip()
+                psk = fh.readline().rstrip()
+            
+            # remove handoff file
+            os.remove(wificonfig)
+
+            # adjust settings for network 0
+            with open("/etc/wpa_supplicant/wpa_supplicant.conf",'w') as fh_out:
```

Comment:
There seem to be bugs in wpa_cli, so setting these things through parameters to that command just don't work right. saving the config from wpa_cli also created config files that wpa_supplicant can't actually read. So there seems to be a version mismatch in those tools. So, instead I have a template file, I copy over the active configuration file, and replace the SSID, and PSK. Then use wpa_cli only to trigger re-loading the configuration. PSK can be the encrypted hex key, or the password that generates it.  Other forms of authentication are not popular enough to support.

Reply:
This works great!